### ISSUE TITLE
Memorization Task

### ISSUE THREAD

**Issue by akul-goyal** [google-research/lm-extraction-benchmark#8]
Hi, can we assume that the prefixes + suffix pairs are memorized by the LM? If that is the case then wouldn't what the LM outputs given a prefix be the correct suffix? Is there a case where the LM has memorized a sequence but upon giving the prefix it does not output the correct suffix?

**Comment by heitikei** [comment]
The logical reasoning for the query to start with "dataset is unknown" for the infinite possible outcomes. This allows for predictive queries to exist in coding. Your thoughts?

**Comment by heitikei** [comment]
Pre- or suf- fix are traditional markers in written language script. They do. not belong in code, it disables the potentiality of data defining best outcomes. Your thoughts are welcome.

**Comment by carlini** [comment]
I'm sorry @heitikei I don't understand what you're trying to say and don't understand how it relates to either machine learning or language model extraction in particular. Can you rephrase your question?

**Comment by heitikei** [comment]
@carlini sorry I am naturally cryptic, looking for the concept in english to explain what I see. Remove identification tags in queries will allow machines, AI and neural networks to perform as imagined......, ? ? ? ?

**Comment by heitikei** [comment]
Reinforcement training is no longer useful

**Comment by akul-goyal** [comment]
@carlini If I understand correctly if instead of feeding in 50 tokens as the prefix, we feed in 50+x tokens then the LM is guaranteed to spit out the next 50-X tokens?

**Comment by heitikei** [comment]
Agreed. Match.

**Comment by carlini** [comment]
@akul-goyal Not exactly. There is a longer sequence (Call it S) that was partitioned to form `(P1 + P2 + Q)`

If you feed `P1 + P2` then you get `Q`. We give you `P2`, but not `P1`. So `P1` is "almost" enough to give you `Q` but not quite.

**Comment by carlini** [comment]
@heitikei Please open a new issue with an any questions you have. As far as I can tell they are not related to this issue here.

**Comment by heitikei** [comment]
New issue. Close #8.

**Comment by carlini** [comment]
No; I will leave #8 open until we resolve the issue with @akul-goyal. You can open a new issue for your questions which appear different.

**Comment by heitikei** [comment]
Thank you. Finding the concept. Sorry, biological unit is slow.

**Comment by akul-goyal** [comment]
sorry, @carlini I just wanted to confirm from your comment you are giving us P2. Because you wrote "P1 is "almost" enough to give you Q but not quite" and I am not sure if that was an error or not

**Comment by akul-goyal** [comment]
Also in the case, we are being given P2, could you tell us how many tokens P1 is? Or does that vary?

**Comment by carlini** [comment]
Sorry yes! I meant P2 there. I've edited it for correctness for future readers.

P1 is length 50 tokens, P2 is length 50 tokens, and Q is length 50 tokens.

**Comment by heitikei** [comment]
there is a problem?

**Comment by akul-goyal** [comment]
Hey @carlini, assuming that the dataset on the ETH Zürich GitHub repo is correct when giving in the preprefix + prefix to the language model, it only correctly guesses the suffix 56% of the time (on the last 1000 samples). Am I doing something wrong? Shouldn't there be a 100% accuracy if the sequence is memorized?

**Comment by heitikei** [comment]
what is the performance indicator? assume guess "ing" out side of trained data set. This is good. Why is this good? Apologies, I am a natural language programmer and think -> speak. I struggle with emotional qualities of good communication. :/

**Comment by akul-goyal** [comment]
Never mind, was not querying the model correctly!
